core/src/main/python/wlsdeploy/util/target_configuration_helper.py

```diff
@@ -0,0 +1,86 @@
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.write('function create_k8s_secret {')
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.write('kubectl -n $NAMESPACE delete secret ${DOMAIN_UID}-$1 --ignore-not-found')
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.write('kubectl -n $NAMESPACE create secret generic ${DOMAIN_UID}-$1 ' +
+                                           '--from-literal=$2=$3')
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.write('kubectl -n $NAMESPACE label secret ${DOMAIN_UID}-$1 ' +
+                                           'weblogic.domainUID=${DOMAIN_UID}')
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.write('}')
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.write("kubectl -n $NAMESPACE delete secret ${DOMAIN_UID}-weblogic-credentials "
+                                           + "--ignore-not-found")
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.write("kubectl -n $NAMESPACE create secret generic "
+                                           +  "${DOMAIN_UID}-weblogic-credentials "
+                                           +   "--from-literal=username=${ADMIN_USER} --from-literal=password=${ADMIN_PWD}")
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.write('kubectl -n $NAMESPACE label secret ${DOMAIN_UID}-weblogic-credentials ' +
+                                           'weblogic.domainUID=${DOMAIN_UID}')
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.write(NL)
+            k8s_create_script_handle.close()
+
+        k8s_create_script_handle = open(k8s_file, "a+")
+        k8s_create_script_handle.write(NL)
+        for property_name in token_dictionary:
+            # AdminPassword handle differently and SecurityConfig.NodeManagerPasswordEncrypted is the short name
+            # which filters out
+            if property_name in [ 'AdminPassword', 'SecurityConfig.NodeManagerPasswordEncrypted']:
+                continue
+            secret_names = property_name.lower().split('.')
+            command_string = "create_k8s_secret %s %s %s " %( '-'.join(secret_names[:-1]), secret_names[-1],
+                                                              "<changeme>")
+            k8s_create_script_handle.write(command_string)
+            k8s_create_script_handle.write(NL)
+
+        k8s_create_script_handle.close()
+
+def format_as_secret(prop_name):
+    """
+    Format as X.Y.A.B  This is the pattern of the name.
+    :param prop_name: property name
+    :return: formatted name
+    """
+    name_lower_tokens = prop_name.lower().split('.')
```

Comment:
I might have missed something in the code that calls this, but isn't this just an attribute name at this point? If so how can it be always be unique. The variable injector makes a unique name for the property.

Reply:
The caller passing an unique name - self.__format_variable_name(location, attribute). I will rename the variable.